packages/flame_3d/lib/src/extensions/quaternion.dart

```diff
@@ -0,0 +1,76 @@
+import 'dart:math';
+
+import 'package:flame_3d/core.dart';
+
+extension QuaternionExtension on Quaternion {
+  List<double> get storage => [x, y, z, w];
+
+  double dot(Quaternion other) {
+    return x * other.x + y * other.y + z * other.z + w * other.w;
+  }
+
+  Quaternion lerp(Quaternion other, double t) {
+    return QuaternionUtils.lerp(this, other, t);
+  }
+
+  Quaternion slerp(Quaternion other, double t, {double epsilon = 10e-6}) {
+    return QuaternionUtils.slerp(this, other, t, epsilon: epsilon);
+  }
+}
+
+final class QuaternionUtils {
+  QuaternionUtils._();
```

Comment:
Interesting that the analyzer didn't say that this should be const

Reply:
I don't think it makes any difference bc you cannot create an instance of this class
the constructor is to prevent the existence of the default constructor